spec/controllers/miq_report_controller/reports/editor_spec.rb

```diff
@@ -174,6 +174,47 @@
         end
       end
     end
+
+    def non_empty_category
+      cat = FactoryGirl.create(:classification, :name => "non_empty", :description => "Has entries")
+      cat.add_entry(:name => "foo", :description => "Foo")
+      cat.add_entry(:name => "bar", :description => "Bar")
+      cat
+    end
+
+    def empty_category
+      FactoryGirl.create(:classification, :name => "empty", :description => "Zero entries")
+    end
+
+    describe "#categories_hash" do
+      it "returns non-empty categories" do
+        non_empty_category
+        empty_category
+        expect(controller.send(:categories_hash)).to eq("non_empty" => "Has entries")
+      end
+
+      it "omits show: false categories" do
+        non_empty_category.update(:show => false)
```

Comment:
> Used methods instead of let because I dislike relying on let's on-demand execution for side effect

Memoized helpers; the reason you would want to use them here is that now you are writing tests like this that require multiple database writes and slowing your tests down. You should either inline what you need in this example, wrap it in a context block and use `before` with an instance var, or (in my opinion) use a memoized helper to change the attributes given to this object on create. There are many ways to avoid what you're doing here.